pybamm/solvers/processed_variable.py

```diff
@@ -425,9 +476,30 @@ def call_2D(self, t, x, r, y, z):
         else:
             if isinstance(second_dim, np.ndarray) and isinstance(t, np.ndarray):
                 second_dim = second_dim[:, np.newaxis]
-
         return self._interpolation_function((first_dim, second_dim, t))
 
+    def get_spatial_scale(self, name, domain=None):
+        "Returns the spatial scale for a named spatial variable"
+        # Different scale in negative and positive particles
+        if domain == "negative particle":
+            name = "r_n"
+        elif domain == "positive particle":
+            name = "r_p"
+
+        # Try to get length scale
+        if name + " [m]" in self.spatial_vars and name in self.spatial_vars:
+            scale = (
+                self.spatial_vars[name + " [m]"] / self.spatial_vars[name]
+            ).evaluate()[-1]
+        else:
+            if self.warn:
+                pybamm.logger.warning(
+                    "No scale set for spatial variable {}. "
+                    "Using default of 1 [m]".format(name)
+                )
+            scale = 1
+        return scale
```

Comment:
maybe models should have a `r_scale`, `x_scale`, etc in the same way that they have a timescale. Probably a separate issue though. It would simplify processed variables and quickplot a lot.

Reply:
Yeah this would definitely be the neatest solution. I've made a new issue for this.